src/ui/Querybox/QueryboxOptionsProcessing.ts

```diff
@@ -0,0 +1,42 @@
+import { Omnibox, IOmniboxOptions } from '../Omnibox/Omnibox';
+import { Querybox, IQueryboxOptions } from './Querybox';
+import { ComponentOptionsModel } from '../../models/ComponentOptionsModel';
+
+export class QueryboxOptionsProcessing {
+  constructor(public owner: Omnibox | Querybox) {}
+
+  private get options() {
+    return this.owner.options;
+  }
+
+  private set options(options: IOmniboxOptions | IQueryboxOptions) {
+    this.owner.options = options;
+  }
+
+  public postProcess() {
+    this.options = { ...this.options, ...this.owner.componentOptionsModel.get(ComponentOptionsModel.attributesEnum.searchBox) };
```

Comment:
Hmm, in this assignment, we are setting a variable `this.options`, but that could also refer to the getter method on line 8? Maybe it would be better to rename the getter to `private getOptions()`?